sympy/utilities/tests/test_wester.py

```diff
@@ -2542,6 +2542,7 @@ def test_W22():
         (-sin(Min(1, u)) + sin(Min(2, u)), True))
 
 
+@XFAIL
 @slow
 def test_W23():
     a, b = symbols('a b', real=True, positive=True)
```

Comment:
I don't know how to fix this test but apparently it only worked before because it depended on incorrect behaviour from `is_extended_real`.

Reply:
This is an issue that we have met before. The filter 'R' in `roots` does not accept expressions like `b/(2*a**2 - 2*b**2)`. I suggest that the filter https://github.com/sympy/sympy/blob/241380082e5eacac2674c5ff8dedd9f16d3d1406/sympy/polys/polyroots.py#L1034
be replaced by something like

    lambda r: all(a.is_extended_real for a in r.as_numer_denom())

Reply:
Thanks! That seems to do it. I'm not sure how exactly that affected the integral but presumably there is some kind of poly test I can add for what this does.

Reply:
Maybe is_extended_real is overkill, is_real would probably suffice. As to a polynomial example, one that has a root like `1/(a - b)` for a real `a` and `b` should work. The actual polynomial in the test has roots

    {b/(2*a**2 - 2*b**2): 1, a/(2*a**2 - 2*b**2): 1, -b/(2*a**2 - 2*b**2): 1, -a/(2*a**2 - 2*b**2): 1}